EDM-2111: added docs for restore and resume (#1612)
flightctl/flightctl: docs/user/device-api-statuses.md

| `Degraded`          | One or more hardware resources or operating system services are reported to be degraded but in a still functional or recovering state. | `!deviceIsDisconnected && !deviceIsRebooting && ∀ r∈{CPU, Memory, Disk}, status.resources[r]∉{Error, Critical} && ∃ r∈{CPU, Memory, Disk}, status.resources[r]∈{Degraded}` |
| `Error`             | One or more hardware resources or operating system services are reported to be in error or critical state.                             | `!deviceIsDisconnected && !deviceIsRebooting && ∃ r∈{CPU, Memory, Disk}, status.resources[r]∈{Error, Critical}` |
| `Rebooting`         | The device is rebooting.                                                                                                               | `!deviceIsDisconnected && deviceIsRebooting` |
| `Offline`           | The device is disconnected from the service but may still be running.                                                                  | `deviceIsDisconnected` |

> | `AwaitingReconnect` | The device is awaiting reconnection after a restore was made.                                                                          | `deviceIsDisconnected` |

Reviewer: ```suggestion
| `AwaitingReconnect` | The device is awaiting reconnection after the system was restored.                                                                          | `deviceIsDisconnected` |
```

> | `ConflictPaused`    | The device is paused because the device reported a newer renderedVersion than the one we have.   | `deviceIsDisconnected` |

Reviewer: ```suggestion
| `ConflictPaused`    | The device is paused because the device reported a renderedVersion not known to the service.   | `deviceIsDisconnected` |
```
